controllers/routemonitor/routemonitor_test.go

```diff
@@ -42,18 +42,20 @@ var _ = Describe("Routemonitor", func() {
 		mockAdder                    *routemonitormocks.MockRouteMonitorAdder
 		mockBlackboxExporter         *routemonitormocks.MockBlackBoxExporter
 		ctx                          context.Context
-
-		update                                helper.MockHelper
-		delete                                helper.MockHelper
-		get                                   helper.MockHelper
-		create                                helper.MockHelper
-		ensureServiceMonitorResourceAbsent    helper.MockHelper
-		ensurePrometheusRuleResourceAbsent    helper.MockHelper
-		shouldDeleteBlackBoxExporterResources helper.MockHelper
-		ensureBlackBoxExporterResourcesAbsent helper.MockHelper
-		ensureBlackBoxExporterResourcesExist  helper.MockHelper
-		ensureFinalizerAbsent                 helper.MockHelper
-
+		mockResourceComparer         *routemonitormocks.MockResourceComparer
+
+		update                                        helper.MockHelper
+		delete                                        helper.MockHelper
+		get                                           helper.MockHelper
+		create                                        helper.MockHelper
+		ensureServiceMonitorResourceAbsent            helper.MockHelper
+		ensurePrometheusRuleResourceAbsent            helper.MockHelper
+		shouldDeleteBlackBoxExporterResources         helper.MockHelper
+		ensureBlackBoxExporterResourcesAbsent         helper.MockHelper
+		ensureBlackBoxExporterResourcesExist          helper.MockHelper
+		ensureFinalizerAbsent                         helper.MockHelper
+		deepEqual                                     helper.MockHelper
```

Comment:
not sure we actually want to keep this as helper.MockHelper as it's only used for the int inside it

maybe change to (and fix corresponding places)
```suggestion
		deepEqualCalledTimes                                     int
```